app/code/Magento/Swatches/Model/SwatchAttributesProvider.php

```diff
@@ -60,9 +66,13 @@ public function provide(Product $product)
 
             $swatchAttributes = [];
             foreach ($configurableAttributes as $configurableAttribute) {
-                if (array_key_exists($configurableAttribute->getAttributeId(), $swatchAttributeCodeMap)) {
-                    $swatchAttributes[$configurableAttribute->getAttributeId()]
-                        = $configurableAttribute->getProductAttribute();
+                // Due to $this->swatchAttributeCodes->getCodes() - return swatch data for every attribute.
+                if ($this->swatchHelper->isSwatchAttribute($configurableAttribute->getProductAttribute())) {
```

Comment:
For more readability, these conditions can be moved to private method with description of intention of extra condition

Reply:
Also, mark this method as deprecated and add comment that this method introduced only for case when customer already has converted attribute.